osu.Game/Overlays/ModSelection/ModSelection.cs

```diff
@@ -0,0 +1,520 @@
+                                // FIXME? probably remove after #202 merged
+                                new Box
+                                {
+                                    Origin = Anchor.BottomLeft,
+                                    Anchor = Anchor.BottomLeft,
+                                    RelativeSizeAxes = Axes.X,
+                                    Height = 5,
+                                    Colour = new Color4(255, 102, 170, 255),
+                                },
+                            }
+                        }
+                    }
+                }
+            };
+            RecreateButtons(PlayMode.Osu);
+        }
+        protected override bool OnKeyDown(InputState state, KeyDownEventArgs args)
+        {
+            switch (args.Key)
+            {
+                case Key.Number1:
+                    ClearMods();
+                    return true;
+                case Key.Escape:
+                case Key.Number2:
+                    if (State == Visibility.Hidden) return false;
+                    Hide();
+                    return true;
+            }
+            return base.OnKeyDown(state, args);
+        }
+        protected override void PopIn()
+        {
+            // FIXME: place actual playmode here
+            if (lastPlayMode != PlayMode.Osu)
+            {
+                RecreateButtons(PlayMode.Osu);
+                lastPlayMode = PlayMode.Osu;
+            }
+
+            MoveToY(0, 500, EasingTypes.OutQuint);
+            FadeIn(500, EasingTypes.OutQuint);
+        }
+
+        protected override void PopOut()
+        {
+            MoveToY(DrawSize.Y, 500, EasingTypes.InSine);
+            FadeOut(500, EasingTypes.InSine);
+        }
+
+        private void ActivateMod(Mod mod)
+        {
+            if (!ActiveTempMods.Contains(mod))
+                ActiveTempMods.Add(mod);
+            else
+                ActiveTempMods.Remove(mod);
+
+            double totalmultiplier = 1.0;
+            bool isRanked = true;
+            foreach (Mod m in ActiveTempMods)
+            {
+                if (m.ScoreMultiplier == 0.0)
+                    isRanked = false;
+
+                totalmultiplier *= m.ScoreMultiplier;
+            }
+            multiplier.Text = totalmultiplier.ToString() + " ";
+
+            if (isRanked)
+                ranked.Text = "Ranked";
+            else
+                ranked.Text = "Unranked";
+        }
+
+        private void ClearMods()
+        {
+            ActiveTempMods.Clear();
+
+            foreach (ModButton button in easymods.Children)
+                button.Disarm();
+
+            foreach (ModButton button in hardmods.Children)
+                button.Disarm();
+
+            foreach (ModButton button in assistmods.Children)
+                button.Disarm();
+
+            multiplier.Text = "1.0 ";
+            ranked.Text = "Ranked";
+        }
+        private void RecreateButtons(PlayMode playmode)

```

Comment:
just call it populateButtons() or something. shouldn't be accepting the playmode here.